test_infra/src/lib.rs

```diff
@@ -798,6 +798,24 @@ pub fn check_matched_lines_count(input: &str, keywords: Vec<&str>, line_count: u
     }
 }
 
+pub fn kill_child(child: &Child) {
+    let r = unsafe { libc::kill(child.id() as i32, libc::SIGTERM) };
+    if r != 0 {
+        let e = io::Error::last_os_error();
+        if e.raw_os_error().unwrap() == libc::ESRCH {
+            return;
+        }
+        eprintln!("Failed to kill child with SIGTERM: {e:?}");
+        let r = unsafe { libc::kill(child.id() as i32, libc::SIGKILL) };
```

Comment:
I don't think `SIGTERM` can fail and `SIGKILL` would succeed. So you can just remove all this

Reply:
Removed.